pkg/types/100/types.go

```diff
@@ -95,6 +96,61 @@ type Result struct {
 	DNS        types.DNS      `json:"dns,omitempty"`
 }
 
+// Note: DNS should be omit if DNS is empty but default Marshal function
+// will output empty structure hence need to write a Marshal function
+func (r *Result) MarshalJSON() ([]byte, error) {
+	var buf bytes.Buffer
```

Comment:
@s1061123 I think a simpler way to do this would be to marshal to JSON, then unmarshal to a `map[string]interface{}`, check if the original DNS is empty and delete it from the map, then re-marshal the map[string]interface{} to a []byte.

That way we don't have to handle each field individually and do a bunch of string manipulation.

```
    bytes, err := json.Marshal(r)
    if err != nil {
        return nil, err
    }

    fixupObj := make(map[string]interface{})
    if err := json.Unmarshal(bytes, &fixupObj); err != nil {
        return err
    }

    if r.DNS.IsEmpty() {
        delete(interimObj, "dns")
    }

    return json.Marshal(interimObj)
```

Reply:
This makes it simple, but this calls json.Unmarshal()/Marshal(). Hence it is a tradeoff between simple code v.s. performance, I guess. But we could choose simple code in such case, because it does not so 'super good' performance. OK. I will change that.

Reply:
Oh, we cannot use 'json.Marshal' in MarshalJSON() because it makes recursion (i.e. json.Marshal(result) -> it calls MarshalJSON() because golang found 'MarshalJSON()' for result -> MarshalJSON() calls json.Marshal(result) again)...

Reply:
I could resolved by type alias. Changed.